### ISSUE TITLE
x/cmd/eg: Clean imports after refactoring

### ISSUE THREAD

**Issue by LOZORD** [golang/go#21328]
tl;dr: While new imports are correctly added by the refactoring, imports that are made obsolete still exist unnecessarily (and therefore __break compilation__). We need to run something like `goimports` after performing the eg transformation.

### What version of Go are you using (`go version`)?
`go version go1.8.1 darwin/amd64`

### What operating system and processor architecture are you using (`go env`)?
darwin/amd64

### What did you do?

```bash
Leos-MBP:~ LeoRudberg$ go build golang.org/x/tools/cmd/eg
Leos-MBP:~ LeoRudberg$ go version
go version go1.8.1 darwin/amd64
Leos-MBP:~ LeoRudberg$ cat test-eg/template.go 
package template

import (
	"errors"
	"fmt"
)

func before(s string) error { return fmt.Errorf("%s", s) }
func after(s string) error  { return errors.New(s) }
Leos-MBP:~ LeoRudberg$ cat test-eg/example.go 
package example

import (
	"fmt"
)

func Whoops(s string) error {
	return fmt.Errorf("%s", "Whoops: "+s)
}
Leos-MBP:~ LeoRudberg$ eg -t test-eg/template.go test-eg/example.go 
=== test-eg/example.go (1 matches)
package example

import (
	"fmt"
	"errors"
)

func Whoops(s string) error {
	return errors.New("Whoops: " + s)
}
```

### What did you expect to see?
The `import` section for the result should only contain `errors`.

### What did you see instead?
The `import` section contained `errors`, but also `fmt` unnecessarily.

**Comment by LOZORD** [comment]
At Gophercon, @alaska started work on this I believe. @clairew also expressed interest in fixing this, too.

**Comment by clairew** [comment]
@LOZORD Thanks for issuing this - will take a look at what I can do!
